File: server/src/learner-activity/core/models/daily-correct-incorrect.model.ts
export class DailyCorrectIncorrect {
  day!: string;
  correct!: number;
  incorrect!: number;

  constructor({ day, correct, incorrect }: ConstructorProps) {
    this.day = String(day);
    this.correct = Number(correct);
    this.incorrect = Number(incorrect);
  }
}

type ConstructorProps = {
  day: string;
  correct: number;
  incorrect: number;
};

File: client/lib/features/statistics/logic/models/daily_correct_incorrect.model.dart
class DailyCorrectIncorrect {
  final String day;
  final int correct;
  final int incorrect;

  DailyCorrectIncorrect({
    required this.day,
    required this.correct,
    required this.incorrect,
  });

  static DailyCorrectIncorrect fromJson(Map<String, dynamic> json) {
    return DailyCorrectIncorrect(
      day: json['day'],
      correct: json['correct'],
      incorrect: json['incorrect'],
    );
  }
}
